### ISSUE TITLE
Crash with Python 3.5.1, OS X 10.11.5

### ISSUE THREAD

**Issue by ssteinerx** [retext-project/retext#224]
Crashing on Python 3.5.1 OS X 10.11.5:

```
$ pip3 install reText
```

```
Collecting reText
  Downloading ReText-6.0.0.tar.gz (177kB)
    100% |████████████████████████████████| 184kB 5.6MB/s
Collecting docutils (from reText)
  Downloading docutils-0.12-py3-none-any.whl (508kB)
    100% |████████████████████████████████| 512kB 2.0MB/s
Collecting Markdown (from reText)
  Downloading Markdown-2.6.6.zip (412kB)
    100% |████████████████████████████████| 419kB 2.3MB/s
Collecting Markups>=2.0 (from reText)
  Downloading Markups-2.0.0.tar.gz
Collecting pyenchant (from reText)
  Downloading pyenchant-1.6.6-py2.py3.cp27.cp26.cp32.cp33.cp34.pp27-none-macosx_10_6_intel.macosx_10_9_intel.whl (3.8MB)
    100% |████████████████████████████████| 3.8MB 434kB/s
Collecting Pygments (from reText)
  Downloading Pygments-2.1.3-py2.py3-none-any.whl (755kB)
    100% |████████████████████████████████| 757kB 1.6MB/s
Installing collected packages: docutils, Markdown, Markups, pyenchant, Pygments, reText
  Running setup.py install for Markdown ... done
  Running setup.py install for Markups ... done
  Running setup.py install for reText ... done
Successfully installed Markdown-2.6.6 Markups-2.0.0 Pygments-2.1.3 docutils-0.12 pyenchant-1.6.6 reText-6.0.0
```

Then, install PyQt5:

```
$ pip3 install PyQt5
```
```
Collecting PyQt5
  Downloading PyQt5-5.6-cp35-cp35m-macosx_10_6_intel.whl (73.2MB)
    100% |████████████████████████████████| 73.2MB 23kB/s
Collecting sip (from PyQt5)
  Downloading sip-4.18-cp35-cp35m-macosx_10_6_intel.whl (46kB)
    100% |████████████████████████████████| 51kB 10.1MB/s
Installing collected packages: sip, PyQt5
Successfully installed PyQt5-5.6 sip-4.18
```

Attempt to run from command line:

```
$ retext
```

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.5/bin/retext", line 27, in <module>
    from ReText.window import ReTextWindow
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/ReText/window.py", line 24, in <module>
    from ReText.tab import ReTextTab, ReTextWebPreview, PreviewNormal, PreviewLive
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/ReText/tab.py", line 32, in <module>
    import enchant
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/enchant/__init__.py", line 92, in <module>
    from enchant import _enchant as _e
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/enchant/_enchant.py", line 121, in <module>
    prefix_dir.contents = c_char_p(e_dir)
TypeError: bytes or integer address expected instead of str instance
```

**Comment by ssteinerx** [comment]
The installation can be fixed by uninstalling pyenchant (which is throwing the exception).

```
$ pip3 uninstall pyenchant 
```

**Comment by mitya57** [comment]
This looks like rfk/pyenchant#45, so not a bug in ReText.

**Comment by ssteinerx** [comment]
If the product has a dependency on the behaviour of a previous version of a tool, but installs a different version of the tool, it's a bug.

**Comment by mitya57** [comment]
That bug happens on macOS only (and there is no fixed version yet). The [wiki section] about Windows and macOS clearly says:

> *Note: these systems are in no way officially supported. Use ReText there at your own risk.*

I may still move enchant to `extras_require` so that it's not pulled in by default, but that would be for other reasons not related to this bug.

[wiki section]: https://github.com/retext-project/retext/wiki/Installing-ReText#for-users-of-windows-and-mac-os-x

**Comment by ssteinerx** [comment]
I'm only suggesting that, if the previous version worked, that the requirements be modified to peg to that version until there's an update that solves this particular problem.

**Comment by mitya57** [comment]
The previous version was released in 2011 and has other bugs.

Let's hope a new version will be released soon (as suggested by rfk/pyenchant@edc6895f0ab733fcc9a0209bb4e11325dc6caacd and https://github.com/rfk/pyenchant/issues/82#issuecomment-223506556).
